src/routes/subscriptions.js

```diff
@@ -0,0 +1,59 @@
+const { StatusCodes } = require('http-status-codes');
+
+const express = require('express');
+const router = express.Router();
+
+// POST /
+// Recieve subscription notification
+router.post('/:id', (req, res) => {
+  const id = req.params.id;
+  res.sendStatus(StatusCodes.OK);
+  const subscription = getSubscription(id);
+  useSubscription(subscription);
+});
+
+router.put('/:id/:resource/:resourceId', (req, res) => {
+  const id = req.params.id;
+  res.sendStatus(StatusCodes.OK);
+  const subscription = getSubscription(id);
+  useSubscription(subscription, req.body);
+});
+function getSubscription(id) {
+  // dummy function for getting the subscription
+  // which can contain information to inform
+  // the app what to do when notified
```

Comment:
I suppose we could POST the subscription to the EHR, check its assigned ID, and then make a PUT to update it to change the endpoint to match so that we could use the `id` to make a request to retrieve the subscription resource.

That seems more involved than using the `id` as a key for the subscription in the database.  Also, the subscription itself is kind of useless.  The only useful information it provides is the criteria, which if we're asking for the resource back, is also not that helpful.

If we want to know what the subscription is actually supposed to DO, our options are to either make unique endpoints that do different things, and point the subscriptions at them, or do what I suggest here and tie each subscription to a specific `action`, which will trigger the correct function.  Both methods seem to be more or less the same.  I thought it would make more sense to include the action separately, instead of as the `id` since if two subscriptions both return to the same endpoint, it would be impossible to retrieve the actual subscription resource either from a database or the EHR.

Reply:
>The only useful information it provides is the criteria, which if we're asking for the resource back, is also not that helpful.

How about instead of trying to get the subscription back we return the information in the notification.
```
{
  "resourceType": "Subscription",
  "id": "1",
  "status": "active",
  "criteria": "Patient?_id=1",
  "channel": {
    "type": "rest-hook",
    "endpoint": "http://**your-ip-here**:3000/notif?action={id}&criteria={subscription criteria - probably need to base64url encode so its not misinterpreted}",
    "payload": "application/fhir+json"
  }
}
```

Reply:
The most important thing we need to do is, if we get a subscription notification, we need to be able to go back and identify which PlanDef/Knowledge Artifact it was created in support of. So that's where maintaining an id is useful. The resource we get back will be mostly useful in identifying the patient & encounter. We don't need to worry about the criteria or Subscription resource because we will have the PlanDef, which is what those were both originally created from.

Reply:
sounds good, in that case it's not really necessary to have an action-id so long as the `id` in the endpoint references a PlanDef.